domain/application/state/state.go

```diff
@@ -161,6 +162,7 @@ func (s *commonStateBase) setCharmState(
 	data := setCharm{
 		UUID:        id.String(),
 		ArchivePath: archivePath,
+		Available:   archivePath != "",
```

Comment:
Is `archivePath` different from `charmStoragePath`?

Reply:
No, they're the same thing.

Reply:
Should we at least name it `storagePath` to have some coherence with the application charm? In any other PR that touches this area

Reply:
It's called an archive path in the sql schema. Storage path is the old state name. We're moving away from the name storage here, only because it implies storage domain.